source/backend/routes/users.js

```diff
@@ -86,4 +88,45 @@ router.delete('/:savedRecipeId', async (req, res) => {
   }
 });
 
+/* POST /auth/signup */
+/* Creates new user in database with username, password and email -> unique user/email 
+required */
+router.post('/auth/signup', async (req, res) => {
+  try{
+    const { username } = req.username;
+    const { password } = req.password;
+    const hashedPwd = await bcrypt.hash(password, 10);
+    const { email } = req.email;
+
+    /* check username and email are unique */
+    const checkUser = await usersModel.getByUsername(username);
+    const checkEmail = await usersModel.getByEmail(email);
+
+    if (checkUser.length != 0) {
+      // not sure if status number is correct, 409 = conflict
+      return res.status(409).json({msg: "Username already exists"})
+    }
+    if ( checkEmail.length != 0) {
+      // not sure if status number is correct, 409 = conflict
+      return res.status(409).json({msg: "E-mail already in use"})
+    }
+
+    /* TODO: create new user */ 
+    const newUser = req.body.user;
+    newUser.password = hashedPwd;
+
+    /* TODO: insert user into database */
+    await usersModel.createUser(newUser);
+
+    return res.status(200).json({ newUser, msg: 'Successfully created a new user' });
+
+  } catch (err) {
+    console.error(err);
+    return res.status(503).json({
+      message: 'Failed to register new user',
+      err,
+    });
+  }
+});
+
 module.exports = router;
```

Comment:
Overall everything looks super good and I think this is nearly ready to merge. Have you been able to test it? I had trouble with testing this in postman